a4c-brooklyn-plugin/src/main/java/alien4cloud/brooklyn/BrooklynProvider.java

```diff
@@ -256,9 +274,12 @@ public void scale(PaaSDeploymentContext deploymentContext, String nodeTemplateId
 
     @Override
     public void getStatus(PaaSDeploymentContext deploymentContext, IPaaSCallback<DeploymentStatus> callback) {
-        Object dep = knownDeployments.get(deploymentContext.getDeploymentId());
-        log.info("GET STATUS - "+dep);
-        if (callback!=null) callback.onSuccess(dep==null ? DeploymentStatus.UNDEPLOYED : DeploymentStatus.DEPLOYED);
+        if (callback!=null) {
+            String entityId = deploymentContext.getDeployment().getOrchestratorDeploymentId();
+            log.info("GET STATUS - " + entityId);
+            Optional<DeploymentStatus> deploymentStatus = deploymentStatuses.get(entityId);
+            callback.onSuccess(!deploymentStatus.isPresent() ? DeploymentStatus.UNDEPLOYED : deploymentStatus.get());
```

Comment:
Not minor enough to block merge: this can be written as `deploymentStatus.or(DeploymentStatus.UNDEPLOYED)`.